### ISSUE TITLE
Bug: updates to the line chart get slower and slower over time

### ISSUE THREAD

**Issue by orwant** [google/google-visualization-issues#1888]
```
What steps will reproduce the problem? Please provide a link to a
demonstration page if at all possible, or attach code.
1.load the attached file in firefox 35.0.1
2.after 2 to 3 minutes you will notice the updates to the chart are taking longer and
longer
3.you will also notice that the memory usage is increasing faster than it should


What component is this issue related to (PieChart, LineChart, DataTable,
Query, etc)? LineChart


Are you using the test environment (version 1.1)? NO
(If you are not sure, answer NO)


What operating system and browser are you using?
Windows 7 Home Premium, Firefox 35.0.1

A week ago I was able to run this page for more than 12 hours without an issue.

I'm not sure my code is using the preferred approach. I have a device I want to query
once a second and chart the data that is returned. The file I attached contains how
I'm updating the chart.

*********************************************************
For developers viewing this issue: please click the 'star' icon to be
notified of future changes, and to let us know how many of you are
interested in seeing it resolved.
*********************************************************

```

Original issue reported on code.google.com by `Chris.Ouellette66` on 2015-02-23 17:21:12

<hr>
- _Attachment: [chart.html](https://storage.googleapis.com/google-code-attachments/google-visualization-api-issues/issue-1888/comment-0/chart.html)_

**Comment by orwant** [comment]
```
Thanks for your report. You may have found a memory leak, which we will investigate.

As a possible workaround, it often helps to clear the chart explicitly between redraws.
I built a jsfiddle from your code that you can try: http://jsfiddle.net/dlaliberte/2gt6t5tt/
```

Original issue reported on code.google.com by `dlaliberte@google.com` on 2015-02-25 14:19:31

**Comment by orwant** [comment]
```
It's great you are looking into the memory leak.

I'm not sure if I should open another issue for this or not but in this issue I also
reported that the chart's updates get slower and slower over time. 

When the code first starts the x-axis divisions of the chart are every second and the
chart is redrawn with each data point added. If you let it run longer, eventually enough
data has been added where the x-axis divisions of the chart are every 10 seconds and
the chart is not redrawn each time gChart.draw() is called but only after 10 seconds.
If you let it run longer till where the x-axis divisions are every minute then the
chart will only be redrawn after a minute. That's 30 calls to gChart.draw().

The old code would redraw the chart with "every" call to gChart.draw().
```

Original issue reported on code.google.com by `Chris.Ouellette66` on 2015-02-25 16:27:03

**Comment by orwant** [comment]
```
I suspect the slowdown is due to the memory leak, but I'll check to see if there is
a disproportionate slowdown even after fixing the memory leak.  Thanks.
```

Original issue reported on code.google.com by `dlaliberte@google.com` on 2015-02-25 16:58:33

**Comment by orwant** [comment]
```
Just to be clear.

"Every" call to gChart.draw() does "not" result in the chart being redrawn. It should
because a new row has been added to the chart data.

My chart example should be updating every 2 seconds. But after running for 2 minutes
you will notice the chart updates once a minute.
```

Original issue reported on code.google.com by `Chris.Ouellette66` on 2015-02-25 17:54:45

**Comment by orwant** [comment]
```
You might be seeing a problem with the timeofday axis that doesn't show the entire range
of values.  If you explicitly set the viewWindow.max to the last value in your data,
that could make a difference.
```

Original issue reported on code.google.com by `dlaliberte@google.com` on 2015-02-25 18:08:22

**Comment by orwant** [comment]
```
thanks, setting viewWindow.max fix the issue with the chart not updating on each call
to gChart.draw()
```

Original issue reported on code.google.com by `Chris.Ouellette66` on 2015-02-25 22:34:05
